File: read_post_api/models/post.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Post struct {
	ID      primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"`
	PostID  int                `json:"post_id,omitempty" bson:"post_id,omitempty"`
	User    string             `json:"user,omitempty" bson:"user,omitempty"`
	Topic   string             `json:"topic,omitempty" bson:"topic,omitempty"`
	Title   string             `json:"title,omitempty" bson:"title,omitempty"`
	Content string             `json:"content,omitempty" bson:"content,omitempty"`
	Created string             `json:"created,omitempty" bson:"created,omitempty"`
}

File: feed_api/src/main/kotlin/com/feedapi/Post.kt
package com.feedapi

import io.micronaut.data.annotation.EmbeddedId
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.serde.annotation.Serdeable

@Serdeable
@MappedEntity
data class Post(
    @field:Id
    var postId: Int,
    var user: String,
    var topic: String,
    var title: String,
    var content: String,
    var created: String,
)
